libraries/apollo-api/src/jsMain/kotlin/com/apollographql/apollo3/api/json/DynamicJsJsonReader.kt

```diff
@@ -0,0 +1,64 @@
+package com.apollographql.apollo3.api.json
+
+
+@Suppress("NOTHING_TO_INLINE")
+private inline fun objectKeys(json: dynamic) = js("Object").keys(json).unsafeCast<Array<String>>()
+
+
+class DynamicMapEntry<K, V>(
+    override val key: K,
+    override val value: V
+) : Map.Entry<K, V>
+
+/**
+ * A [JsonReader] that reads data from a regular JS `dynamic` object.
+ *
+ * Dynamic values should be any of:
+ * - String
+ * - Int
+ * - Double
+ * - Long
+ * - JsonNumber
+ * - null
+ * - `dynamic` where values are any of these values recursively
+ * - Array<Any?> where values are any of these values recursively
+ *
+ * Anything else is undefined
+ *
+ * TODO Note: We can probably safely remove the non-JS primitives (ie Long, Double, JsonNumber)
```

Comment:
I think JS objects support those (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Grammar_and_types#data_structures_and_types). Kotlin seems to map JsonNumber (our BigInt) to `Double` but I guess this might change.

```kotlin
       val o = js("{ float: 3.14, int: 3.0, bigint: 922337203685477580700 }")

        println(o.float::class.simpleName) // "Double"
        println(o.int::class.simpleName) // "Int"
        println(o.bigint::class.simpleName) // "Double"
```

Reply:
Oh, interesting. Ok! I can remove the TODO.